tests/test_buffer_issue.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+import pyfive
+import s3fs
+
+
+def _load_nc_file(ncvar):
+    """
+    Get the netcdf file and its b-tree.
+
+    Fixture to test loading an issue file.
+    """
+    issue_file = "da193a_25_6hr_t_pt_cordex__198807-198807.nc" 
+    storage_options = {
+        'anon': True,
+        'client_kwargs': {'endpoint_url': "https://uor-aces-o.s3-ext.jc.rl.ac.uk"},  # final proxy
+    }
+    test_file_uri = os.path.join(
+        "esmvaltool-zarr",
+        issue_file
+    )
+    fs = s3fs.S3FileSystem(**storage_options)
+    s3file = fs.open(test_file_uri, 'rb')
+    nc = pyfive.File(s3file)
+    ds = nc[ncvar]
+
+    return ds
+
+
+def test_buffer_issue():
+    """
+    Test the case when the attribute contains no data.
+
+    This happens when DATASPACE is NULL and DATA is empty.
+    """
+    print("File with issue da193a_25_6hr_t_pt_cordex__198807-198807.nc")
+    print("Variable m01s30i111")
+    _load_nc_file('m01s30i111')
```

Comment:
@kmuehlbauer in lieu of an actual small file I've added a test case for an S3-hosted file - this works beautifully with this PR

Reply:
Nice!

Reply:
if you ready with this PR, I reckon we can merge, bud. We should definitely prepare a new release with this and #105